hadoop-ozone/ozone-recon/src/main/java/org/apache/hadoop/ozone/recon/api/ContainerKeyService.java

```diff
@@ -72,10 +74,11 @@
    * @return {@link Response}
    */
   @GET
-  public Response getContainers() {
+  public Response getContainers(
+      @DefaultValue("-1") @QueryParam("limit") int limit) {
     Map<Long, ContainerMetadata> containersMap;
     try {
-      containersMap = containerDBServiceProvider.getContainers();
+      containersMap = containerDBServiceProvider.getContainers(limit);
```

Comment:
Default value of the query param limit is -1.
So, in the case of -1, it behaves like normal, it lists all the containers.

Reply:
Yes, that's correct.

Reply:
Okay. Then I feel we can have 2 methods, getContainers(), getContainers(int limit)
In this way, we don't need any special considerations in underlying implementation